src/GafferUI/ViewportGadget.cpp

```diff
@@ -1216,8 +1218,11 @@ bool ViewportGadget::buttonRelease( GadgetPtr gadget, const ButtonEvent &event )
 
 bool ViewportGadget::buttonDoubleClick( GadgetPtr gadget, const ButtonEvent &event )
 {
-	/// \todo Implement me. I'm not sure who this event should go to - probably
-	/// the last button press gadget, but we erased that in buttonRelease.
+	if( m_previousClickGadget )
+	{
+		return dispatchEvent( m_previousClickGadget.get(), &Gadget::buttonDoubleClickSignal, event );
```

Comment:
Trying to improve the special case where a double click actually happens doesn't seem that important - the common case is single clicks, in which case we hold on m_previousClickGadget indefinitely waiting for a double that never comes.  If it's any sort of problem, it seems like we need to deal with the common case?

As it is, it feels a bit more consistent to just let m_previousClickGadget always stick around, rather than clearing it rarely.